spec.bs

```diff
@@ -70,6 +70,66 @@ dictionary AuctionAdInterestGroup {
+    1. [=list/For each=] |ad| in |group|["{{AuctionAdInterestGroup/ads}}"]:
+        1. Let ig_ad be a new [=interest group ad=]. If |ad|["{{AuctionAd/renderUrl}}"] is a valid [=URL=], then set |ig_ad|'s [=interest group ad/render url=] to |ad|["{{AuctionAd/renderUrl}}"]. Otherwise, return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+        1. If |ad|["{{AuctionAd/metadata}}"] is not null, let |ig_ad|'s [=interest group ad/metadata=] be the result of [=JSON-serializing=] |ad|["{{AuctionAd/metadata}}"] into a string. If it fails, then return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+        1. [=list/Append=] |ig_ad| to |interestGroup|'s [=interest group/ads=].
+    1. [=list/For each=] |ad| in |group|["{{AuctionAdInterestGroup/adComponents}}"]:
+        1. Let ig_ad be a new [=interest group ad=]. If |ad|["{{AuctionAd/renderUrl}}"] is a valid [=URL=], then set |ig_ad|'s [=interest group ad/render url=] to |ad|["{{AuctionAd/renderUrl}}"]. Otherwise, return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+        1. If |ad|["{{AuctionAd/metadata}}"] is not null, let |ig_ad|'s [=interest group ad/metadata=] be the result of [=JSON-serializing=] |ad|["{{AuctionAd/metadata}}"] into a string. If it fails, return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+        1. [=list/Append=] |ig_ad| to |interestGroup|'s [=interest group/ad components=].
+1. Validate the new interest group |interestGroup|:
+    1. If |interestGroup|'s [=interest group/owner=] [=url/scheme=] is not HTTPS, then return false.
+    1. If |interestGroup|'s [=interest group/priority=] is infinite, then return false.
+    1. For each |url| in |interestGroup|'s [=interest group/bidding url], [=interest group/bidding wasm helper url=], [=interest group/daily update url=], [=interest group/trusted bidding signals url=], return false if |url|:
+        1. is not [=same origin=] with |interestGroup|'s [=interest group/owner=].
+        1. is not a valid [=URL=].
+        1. [=url/scheme=] is not HTTPS.
+        1. has non-empty <a href="https://url.spec.whatwg.org/#concept-url-username">username</a> or <a href="https://url.spec.whatwg.org/#concept-url-password">password</a>.
+        1. <a href="https://url.spec.whatwg.org/#concept-url-fragment">fragment</a> is not null.
+    1. [=list/For each=] |ad| in |interestGroup|'s [=interest group/ads=] (same for |interestGroup|'s [=interest group/ad components=]), return false if |ad|'s [=interest group ad/render url=]:
+        1. is not a valid [=URL=].
+        1. [=url/scheme=] is not HTTPS.
+        1. has non-empty <a href="https://url.spec.whatwg.org/#concept-url-username">username</a> or <a href="https://url.spec.whatwg.org/#concept-url-password">password</a>.
+    1. If |interestGroup|'s [=estimate size of an interest group|estimated size=] is greater than 50 KB, return false.

```

Comment:
@jyasskin We're wondering if this size limit is more of a FLEDGE implementation detail, or it goes to the spec as well. Any thoughts? Thanks!